options.hpp

```diff
@@ -35,12 +35,24 @@ class database_options_t {
     std::string conninfo() const;
 };
 
+class table_options_t {
+public:
+    table_options_t();
+    int hstore_mode; ///< add an additional hstore column with objects key/value pairs, and what type of hstore column
+    bool hstore_match_only; ///< only copy rows that match an explicitly listed key
+    std::vector<std::string> hstore_columns; ///< list of columns that should be written into their own hstore column
+    bool enable_hstore_index; ///< add an index on the hstore column
+    boost::optional<std::string> tblsmain_index; ///< Pg Tablespace to store indexes on main tables (no default TABLESPACE)
+    boost::optional<std::string> tblsmain_data; ///< Pg Tablespace to store main tables (no default TABLESPACE)
```

Comment:
These two should be renamed into table_space_index/data as this class as defined is for tables and general and the 'main' is just confusing.

Two other personal preferences (ignore at will): make the class a struct (because of the public members) and use default member initializers instead of a default constructor(to keep the code compact).